src/transformers/models/t5gemma/modeling_t5gemma.py

```diff
@@ -341,7 +342,14 @@ def forward(
             value_states = self.v_proj(encoder_hidden_states).view(encoder_hidden_shape).transpose(1, 2)
 
             if past_key_values is not None:
-                key_states, value_states = curr_past_key_value.update(key_states, value_states, self.layer_idx)
+                # For cross attention with static cache, we need to cache all encoder positions at once
```

Comment:
This change was needed to pass the unit tests - open to reverting / changing it!